code/modules/research/machinery/_production.dm

```diff
@@ -43,6 +43,9 @@
 	RefreshParts()
 	update_icon(UPDATE_OVERLAYS)
 
+	RegisterSignal(stored_research, COMSIG_TECHWEB_ADD_DESIGN, .proc/update_designs)
+	RegisterSignal(stored_research, COMSIG_TECHWEB_REMOVE_DESIGN, .proc/update_designs)
```

Comment:
Does update_designs have SIGNAL_HANDLER?

Reply:
I'm blind, I'll get that in :)